### PR TITLE
Add toggle to disable ssl key handling

### LINKED ISSUE #479: Unnecessary file declaration in "puppet-puppet/manifests/server/config.pp"
On line 142 of "puppet-puppet/manifests/server/config.pp" starts the declaration of the puppetserver cert private key. The file is defined without content, and I don't see the possibility to hand over content to the module. So as far as I see it, this only gives me a "Duplicate declaration" error if I try to fill the file outside of the module.
Sadly that's currently the showstopper for my role-out of the module, and I would be very grateful for any hint on how to make this part of the module work.

```
  file { "${::puppet::server::ssl_dir}/private_keys/${::puppet::server::certname}.pem":
    owner => $::puppet::server::user,
    group => $::puppet::server::group,
    mode  => '0640',
  }
```

### PR DESCRIPTION
**Pull request by preisbeck**
fixes #479 
To enable external certifacte and key handling add a new boolean to disable the standard private_key

### CODE REVIEW

File: spec/classes/puppet_server_config_spec.rb [hunk 1]
```diff
@@ -630,6 +630,21 @@
...
+          "class {'puppet':
+            server                => true,
+            server_certname       => 'servercert',
+            server_ssl_key_manage => false,
+            server_ssl_dir        => '/etc/custom/puppetlabs/puppet/ssl'
+          }"
+        end
+
+        it 'should not contain ssl key configuration in the config section' do
+          should_not contain_file('/etc/custom/puppetlabs/puppet/ssl/private_keys/servercert.pem')
```
Comment (mmoll) [inline spec/classes/puppet_server_config_spec.rb]: the test doesn't match the description

**Comment by mmoll** [comment]
one comment, looking good otherwise

**Comment by preisbeck** [comment]
I updated the test description.

**Comment by ekohl** [comment]
Thanks!

**Pull request merged**